### PR TITLE
feat(chat): enable Claude Remote Control mid-turn via deferred queue

### PR DESCRIPTION
**Pull request by jamesbrink**
## Summary

- Lets the user click **Claude Remote Control → Enable** in the composer overflow menu while a turn is in flight. The click queues a pending-enable intent and a deferred-fire effect applies it the moment the turn ends (or is stopped). Previously the entire menu was locked mid-turn, forcing the user to wait, click Stop, or send the message without remote control armed.
- The pending state lives in `OverflowMenu` (parent), not `RemoteControlMenuItem` (dropdown child), so it survives closing the dropdown — clicking outside or pressing Escape no longer discards the queued intent.
- Other menu items (Fast Mode, Claude in Chrome, ModelSelector, Plan, ReasoningPill) keep their pre-existing mid-turn lock since they would re-spawn / reconfigure the live persistent session.

## Why

Today the overflow trigger and every row are disabled whenever the turn is running, because the rows that *do* mutate session state (Fast Mode, Chrome) can't safely fire mid-turn. Remote Control doesn't have that constraint — the backend's enable RPC just stages a bridge for the next turn — but it shared the lock by accident. The result was that anyone realising "I want to drive this from my phone in a sec" mid-turn had no way to arm it; they'd send the next message, *then* enable, *then* send a second message.

### Behavior

- **`ChatInputArea.tsx`** — split the `disabled` prop into `disabled` (workspace environment preparing — hard-blocks everything) and `isRunning` (turn in flight — soft-locks session-mutating rows only).
- **`ComposerToolbar.tsx`** — accepts both props; internally computes `mutationDisabled = disabled || isRunning` for the pills (ModelSelector, Plan, ReasoningPill) so their pre-split lock is preserved.
- **`OverflowMenu.tsx`** — trigger button only disabled by the hard \`disabled\`; `mutationDisabled` gates Fast Mode and Chrome MenuItems. Remote Control row stays clickable. Pending-enable state, \`runImmediate\`, session-switch cleanup effect, and deferred-fire effect all owned by `OverflowMenu` so they survive the dropdown unmount.
- **`OverflowMenu.module.css`** — `.remoteGroupPending` chip styling and disabled `.item` styling.

### Click semantics for Remote Control (new \`toggle\` body)

1. **`busy`** (real `enabling` state, not the pending intent) — ignore. Matches pre-change behaviour.
2. **Re-click while pending** — cancel the queued intent; toast says "Remote Control enable cancelled."
3. **Enable while a turn is running** — queue the pending intent. Row shows a "pending" chip plus "Pending — will enable when the current turn finishes." Toast confirms.
4. **Idle, or any disable click** — fire immediately. Disable mid-turn is allowed without waiting (tearing down a bridge doesn't need turn quiescence).

### Pending intent keyed by sessionId

\`pendingEnableForSession\` is `string | null` (the sessionId, or null) rather than a plain boolean. A session prop change in the same render pass can't accidentally fire the queued enable against the new session before the cleanup effect wipes it. Switching chats mid-pending clears the intent.

### Deferred-fire seam

The effect watches `isRunning` and the pending sessionId:

```ts
useEffect(() => {
  if (pendingEnableForSession !== sessionId) return;
  if (isRunning) return;
  setPendingEnableForSession(null);
  void runImmediate(true);
}, [pendingEnableForSession, sessionId, isRunning, runImmediate]);
```

Piggybacks on the existing \`agent_status\` transition that \`useAgentStream.ts\` pushes into the store at the end of every turn. No new Tauri event, no backend hook. Also fires for the "Stopped" path (user-cancelled turn), since the persistent session is no longer mid-flight either way.

### No-op-message safety net (documented, not wired)

The backend's cold-enable defer (\`should_defer_enable_until_first_turn\` in \`remote_control.rs\`) can return \`state: "enabling"\` with detail "Send your first message to start Claude Remote Control." when called on a chat with \`turn_count == 0\`. That branch is unreachable from the mid-turn-pending path (queuing requires \`isRunning === true\`, which requires \`turn_count >= 1\`), so no synthetic message is sent. The seam is documented in the deferred-fire effect's comment for future flows.

## Bug fix included (second commit)

Initial implementation kept the pending state inside `RemoteControlMenuItem`. That component only mounts while the dropdown is open, so a click outside the menu unmounted the row and discarded the queued intent — the turn would end and nothing fired. Caught by manual UAT after the first push, fixed by hoisting state + effect into `OverflowMenu` (which stays mounted as long as the composer does). Regression pinned with a test that queues, dispatches a `mousedown` on `document.body`, then rerenders with \`isRunning=false\` and asserts the enable RPC fires.

Corner cases still escape this scope and are noted in the source comment:
- Full window reload mid-turn loses the queued intent.
- Opening a file/diff that unmounts `<ChatPanel>` loses it too.

Both would require promoting state to a Zustand slice. Not done here.

## Docs

- \`site/src/content/docs/features/claude-remote-control.mdx\` — added the "click Enable at any time" paragraph, new **Pending** row in the lifecycle table, armed-vs-pending disambiguation in Troubleshooting.

## Test plan

- [x] \`bunx tsc -b\` — clean
- [x] \`bun run lint\` — 0 errors (16 pre-existing warnings in unrelated files)
- [x] \`bun run lint:css\` — clean
- [x] \`bun run test\` — **2487 passed, 0 failed**
  - 11 OverflowMenu tests, including:
    - Mid-turn trigger reachability (regression vs. the old hard lock)
    - Fast Mode + Chrome stay disabled mid-turn (no session-mutation drift)
    - Queue → defer-fire on \`isRunning\` flip
    - Re-click cancels queued intent
    - Idle path fires immediately (regression vs. existing behaviour)
    - Session switch clears the pending intent
    - **Dropdown close → reopen → still fires** (regression for the second-commit bug)
- [x] \`cargo clippy -p claudette -p claudette-server -p claudette-cli --all-targets --all-features\` with \`RUSTFLAGS=-Dwarnings\` — clean. Zero \`.rs\` files modified, so existing \`remote_control\` backend tests cannot have regressed by construction.
- [x] Manual UAT in the dev build — queue mid-turn, close the menu, wait for turn end → Remote Control lights up. Cancel-by-re-click works. Idle enable unchanged. Session switch wipes the queue.

## Files

| File | Purpose |
|---|---|
| \`src/ui/src/components/chat/ChatInputArea.tsx\` | Split \`disabled\` into \`disabled\` + \`isRunning\` |
| \`src/ui/src/components/chat/composer/ComposerToolbar.tsx\` | Accepts both, computes \`mutationDisabled\` locally for pills |
| \`src/ui/src/components/chat/composer/OverflowMenu.tsx\` | Trigger reachable mid-turn; owns pending state, \`runImmediate\`, and deferred-fire effect |
| \`src/ui/src/components/chat/composer/OverflowMenu.module.css\` | \`.remoteGroupPending\` chip + disabled \`.item\` styling |
| \`src/ui/src/components/chat/composer/OverflowMenu.test.tsx\` | New: 11 tests covering reachability, queue/cancel/defer, dropdown-close regression, session-switch isolation |
| \`site/src/content/docs/features/claude-remote-control.mdx\` | Pending lifecycle row + armed-vs-pending Troubleshooting note |

### CODE REVIEW

File: src/ui/src/components/chat/composer/OverflowMenu.tsx [hunk 1]
```diff
@@ -61,6 +77,120 @@ export function OverflowMenu({ sessionId, disabled, isRemote }: OverflowMenuProp
...
+  // reach that branch — queuing the pending intent requires
+  // `isRunning === true`, which in turn requires a turn to be
+  // running, which requires `turn_count >= 1` and a live persistent
+  // session.
+  useEffect(() => {
+    if (pendingEnableForSession !== sessionId) return;
+    if (isRunning) return;
+    setPendingEnableForSession(null);
+    void runImmediate(true);
+  }, [pendingEnableForSession, sessionId, isRunning, runImmediate]);
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/composer/OverflowMenu.tsx]: Addressed in e4ae4b0c. Good catch — the gate-off race was real. Fix has two parts:

1. New effect wipes `pendingEnableForSession` when `showRemoteControl` flips false (the experimental gate or the remote-transport branch). Honours the user's decision to turn the feature off instead of silently honouring a stale queued intent.
2. Belt-and-suspenders guard on the deferred-fire effect — it now bails on `!showRemoteControl` and on `disabled` (workspace preparing) before invoking the RPC.

Pinned by a new regression test: queue mid-turn, flip the gate off, end the turn, assert `setClaudeRemoteControl` stayed silent.

File: src/ui/src/components/chat/composer/OverflowMenu.test.tsx [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+    expect(() => findRemoteControlButton(container)).not.toThrow();
+  });
+});
+
+// --- End-to-end toggle behavior ----------------------------------------
+//
+// These exercise the user-implemented `toggle()` callback per the sketch
+// in `OverflowMenu.tsx`. They will fail against the placeholder body
+// (which fires immediately) and pass once the queue / cancel / defer
+// branches are wired up. Each test names the exact branch it covers.
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/composer/OverflowMenu.test.tsx]: Addressed in e4ae4b0c. The stale "user-implemented `toggle()` callback" / "placeholder body" language was a leftover from the learning-mode scaffolding workflow — the toggle was a stub when those tests were authored, then the stub got filled in but the section header didn't. Rewrote the header to describe what the tests actually cover (queue, cancel, defer-fire, idle regression, dropdown-close-survival regression).
